README.md

```diff
@@ -15,6 +15,55 @@ Our documentation covers:
 * [Failures that Kani can spot](https://model-checking.github.io/kani/tutorial-kinds-of-failure.html)
 * [Kani's current limitations](https://model-checking.github.io/kani/limitations.html)
 
+## What guarantees does Kani provide?
```

Comment:
I think the Kani front-page is not the best place for this information. Can we move it to a section in the documentation?